editor/export/project_export.cpp

```diff
@@ -455,6 +478,75 @@ void ProjectExportDialog::_tab_changed(int) {
 	_update_feature_list();
 }
 
+void ProjectExportDialog::_patch_button_clicked(Object *p_item, int p_column, int p_id, int p_mouse_button_index) {
+	TreeItem *ti = Object::cast_to<TreeItem>(p_item);
+
+	patch_index = ti->get_metadata(0);
+
+	Ref<EditorExportPreset> current = get_current_preset();
+	ERR_FAIL_COND(current.is_null());
+
+	if (p_id == 0) {
+		Vector<String> preset_patches = current->get_patches();
+		ERR_FAIL_INDEX(patch_index, preset_patches.size());
+		patch_erase->set_text(vformat(TTR("Delete patch '%s' from list?"), preset_patches[patch_index].get_file()));
+		patch_erase->popup_centered();
+	} else {
+		patch_dialog->popup_centered_ratio();
+	}
+}
+
+void ProjectExportDialog::_patch_edited() {
+	TreeItem *item = patches->get_edited();
+	if (!item) {
+		return;
+	}
+	int index = item->get_metadata(0);
+
+	Ref<EditorExportPreset> current = get_current_preset();
+	ERR_FAIL_COND(current.is_null());
+
+	Vector<String> preset_patches = current->get_patches();
+	ERR_FAIL_INDEX(index, preset_patches.size());
+
+	current->set_patch(index, preset_patches[index]);
+}
+
+void ProjectExportDialog::_patch_added() {
```

Comment:
This method is named wrongly. It's called before adding a patch (when button is pressed).
`_add_pack_pressed` would be more accurate.